### DISCUSSION TITLE
A little help please

### DISCUSSION THREAD

**Discussion by Just-another-bloke** [TomBursch/kitchenowl#66] [Q&A]
Hi, Sorry I am fairly new to Linux, Dockers and all that jazz, learning fast but not fast enough it seems.

I have deployed KitchenOwl but when I browse to the front end I get "hmmmmm... Couldn't reach server" When I expose the port for the backend I get "Requested resource not found"

I am using Portainer, the default compose from your website but with the config path changed for my environment. 

Not sure what I am doing wrong but it looks like the backend isnt firing up. I can post logs but they may be a bit big for here. 

Appreciate any pointers I can get.

Thanks in advance for any help

Danny

**Comment by jfdid** [comment]
Hi Danny,
As far as I know, if you're getting the "hmm.." message, your port is open and accessible.
What is your FRONT_URL in your docker-compose?  And where are you trying to access it from?
If your host has a static IP, I'd suggest setting the FRONT_URL and BACK_URL to the machine's IP rather than "localhost".  
If you set it to the machine's IP (eg 192.168.1.5), you NEED to access it by entering that address into the address bar - if you use localhost or 0.0.0.0 in the address bar, it causes issues.
Let me know more about your setup and we should be able to get it going - you're almost there by the looks of things.

**Reply by Just-another-bloke** [reply]
Hi there,

Thanks for the quick reply, The Front URL is setup to be a hostname on a domain I own, that bit is working, I can access the front end but it comes back with the hmmm. The host is a static IP and my internal DNS is pointing the URL to that server, externally as well its published through reverse proxy on my opnsense firewall.

I haven't set a back URL, I could do but I think the issue is more the backend isn't coming up clean, I added the below to the "back" config so I could see what is happening on the back website and I think that's what is not working.

        ports:
        - "7181:80"
when I browse to http://192.168.0.230:7181 I get "Requested resource not found" which is probably the default message the web server spits out when its broken

My compose looks like this

version: "3"
services:
    front:
        image: tombursch/kitchenowl-web:latest
        environment:
        - FRONT_URL=http://external.mydomain.co.uk
        ports:
        - "7180:80"
        depends_on:
        - back
        networks:
        - default
    back:
        image: tombursch/kitchenowl:latest
        restart: unless-stopped
        networks:
        - default
        ports:
        - "7181:80"
        environment:
        - JWT_SECRET_KEY=Changedthisonebutnotsurewhereitisrelevant?
        - FRONT_URL=http://external.mydomain.co.uk
        volumes:
        - /mnt/1TB/config/kitchenowl_data:/data

volumes:
    data:

networks:
    default:

Also I can see when I run it that the only data in the kitchenowl_data folder is an upload folder and a database.db should there be more in there?

Thanks again

Danny

**Comment by jfdid** [comment]
Hi Danny,
Try removing your custom domain and swapping it in for the machine's address 192.168.0.230 just to test and see if that works.  If so, it's probably the same issue as others where I think there's an ENV that will be fixed on the next release.  When Tom says the FRONT_URL has to be exactly what you use to access the server, he means EXACTLY(even though localhost/domain forwards/static address mapping all work in usual circumstances, not here yet).
And the data folder only has database.db and upload on my side as well - I'm assuming that's normal because mine is working great.
Report back to see if that above test worked :)

**Reply by Just-another-bloke** [reply]
No joy my friend, same error. I will keep plugging away on this side later, its time to go get the kids from School, I think its the back end, maybe something in my deployment it doesn't like as I am sure its not running correctly.

Scratch that, I just tried one more time and it worked this time, I think I messed up the previous deployment while trying to mess around with the ports. Now I have a login screen. Looks like the IP did the trick mate, Appreciate the assistance and have a great weekend

Thanks
Danny

**Comment by jfdid** [comment]
Happy to help - I just discovered the project the other day and think it's great so I'm trying to help where I can!
As far as the domain and reverse proxy, let us know if you try that again - it may not work until the new release but you could try a simple port forward while leaving your docker-compose as-is with the static IP.  I'm not a reverse proxy guy so I won't be much help here but someone else may chime in.

**Comment by Just-another-bloke** [comment]
No joy on the reverse proxy yet, will figure something out. 

I saw the project and thought it would be perfect, my wife is always telling me to go shopping and I am always forgetting stuff. This seemed like the perfect solution, she can add stuff as she remembers it and when I go I can just open it up and get everything. Win win.

Eventually we can get all our recipes in there too so it's a one stop shop

Thanks again for your help mate, much appreciated

Danny

**Reply by TomBursch** [reply]
If you're using ssl on your reverse proxy the schema has to be set correctly as well, so `https://external.mydomain.co.uk/` instead of `http://external.mydomain.co.uk/`

> when I browse to http://192.168.0.230:7181/ I get "Requested resource not found" which is probably the default message the web server spits out when its broken

That's expected, try `http://192.168.0.230:7181/api/health/8M4F88S8ooi4sMbLBfkkV7ctWwgibW6V/` and it should give a valid response

**Reply by Just-another-bloke** [reply]
Thanks Tom, that makes sense and I think I have found the issue using the URL instead of the IP, I recently swapped over to pi-hole for my DNS and the server was pointed at that, my AD DNS was being bypassed.

I will have a play with the configuration later but now I have got this far I can work out the rest I think.

Thank you both for the help and advice.

Thanks

Danny

**Comment by jfdid** [comment]
Exactly my use-case as well - it's awesome.
The recipe import function actually works very well too - haven't moved a lot of recipes over here yet but it'll be helpful when I do.
